python/dgl/data/movielens.py

```diff
@@ -0,0 +1,646 @@
+"""MovieLens dataset"""
+import os
+import re
+import zipfile
+
+import numpy as np
+import pandas as pd
+
+from ..backend import data_type_dict, tensor
+
+from ..convert import graph
+from ..transforms.functional import add_reverse_edges
+from .dgl_dataset import DGLDataset
+
+from .utils import (
+    download,
+    load_graphs,
+    load_info,
+    save_graphs,
+    save_info,
+    split_dataset,
+)
+
+GENRES_ML_100K = [
+    "unknown",
+    "Action",
+    "Adventure",
+    "Animation",
+    "Children",
+    "Comedy",
+    "Crime",
+    "Documentary",
+    "Drama",
+    "Fantasy",
+    "Film-Noir",
+    "Horror",
+    "Musical",
+    "Mystery",
+    "Romance",
+    "Sci-Fi",
+    "Thriller",
+    "War",
+    "Western",
+]
+GENRES_ML_1M = GENRES_ML_100K[1:]
+GENRES_ML_10M = GENRES_ML_100K + ["IMAX"]
+
+
+class MovieLensDataset(DGLDataset):
+    r"""MovieLens dataset.
```

Comment:
Is there a paper link?

Reply:
Yes. I add reference and related website link in the latest commit.

Reply:
Follow the practice in https://github.com/dmlc/dgl/pull/5511

Reply:
fix in the latest commit